src/Hedgehog/Report.fs

```diff
@@ -0,0 +1,106 @@
+namespace Hedgehog
+
+[<Measure>] type tests
+[<Measure>] type discards
+[<Measure>] type shrinks
+
+type FailureData = {
+    Size : Size
+    Seed : Seed
+    Shrinks : int<shrinks>
+    Journal : Journal
+    RenderRecheck : bool
+}
+
+type Status =
+    | Failed of FailureData
+    | GaveUp
+    | OK
+
+type Report = {
+    Tests : int<tests>
+    Discards : int<discards>
+    Status : Status
+}
+
+module Report =
+    open System
+    open System.Text
+
+    let private renderTests : int<tests> -> string = function
+        | 1<tests> ->
+            "1 test"
+        | n ->
+            sprintf "%d tests" n
+
+    let private renderDiscards : int<discards> -> string = function
+        | 1<discards> ->
+            "1 discard"
+        | n ->
+            sprintf "%d discards" n
+
+    let private renderAndDiscards : int<discards> -> string = function
+        | 0<discards> ->
+            ""
+        | 1<discards> ->
+            " and 1 discard"
+        | n ->
+            sprintf " and %d discards" n
+
+    let private renderAndShrinks : int<shrinks> -> string = function
+        | 0<shrinks> ->
+            ""
+        | 1<shrinks> ->
+            " and 1 shrink"
+        | n ->
+            sprintf " and %d shrinks" n
+
+    let private append (sb : StringBuilder) (msg : string) : unit =
+        sb.AppendLine msg |> ignore
```

Comment:
How about calling this `appendLine` to match the wrapped method?

Reply:
Done, also renamed `renderf` to `appendLinef`